src/ui/src/components/core/content/CoreChatBot/CoreChatbotMessage.vue

```diff
@@ -107,6 +183,30 @@ const contentBgColor = computed(() => {
 			return "";
 	}
 });
+
+const handleImageError = (event: Event) => {
+	const img = event.target as HTMLImageElement;
+	img.style.display = "none";
+};
+
+const showImageModal = ref(false);
+const modalImageUrl = ref("");
+
+const openImageModal = (url?: string) => {
+	if (url) {
+		modalImageUrl.value = url;
+		showImageModal.value = true;
+	}
+};
+
+const closeImageModal = () => {
+	showImageModal.value = false;
+	modalImageUrl.value = "";
+};
```

Comment:
From my previous comment, you can simply use `modalImageUrl`, and display the image if this ref is filled.
```suggestion
const modalImageUrl = ref<string | undefined>();
```